### PR TITLE
feat: Integrate TLDraw with BigBlueButton (part 1)

### PR DESCRIPTION
**Pull request by KDSBrowne**
### What does this PR do?
This PR integrates the [TLDraw ](https://github.com/tldraw) whiteboard into the html5 client.

BBB-WB-TLDRAW

### More
This is just a draft PR as it is still a work in progress while we transitioning from the POC to proper integration with akka-apps. 

There are still some issues which will be addressed in near future:
- [x] Deleting last whiteboard shape doesn't update for other users.
- [x] Restore presentation toolbar.
- [x] Control slide changes with previous `changeSlide` / `currentPage`.
- [ ] Button to enable and disable multiuser whiteboard for all users.
- [x] Disable read only mode when multiuser enabled.
- [x] Exception while invoking method persistAsset MongoServerError: After applying the update.
- [x] Following presenter's view via akka.
- [x] Remove hard coded slide dimensions.

### CODE REVIEW

File: bigbluebutton-html5/imports/api/audio/client/bridge/sip.js [hunk 1]
```diff
@@ -482,7 +484,7 @@ class SIPSession {
         },
         displayName: callerIdName,
         register: false,
-        userAgentString: 'BigBlueButton',
+        userAgentString: `BigBlueButton/${UA_SERVER_VERSION} (HTML5, rv:${UA_CLIENT_VERSION}) ${window.navigator.userAgent}`,
```
Comment (antobinary) [inline bigbluebutton-html5/imports/api/audio/client/bridge/sip.js:487]: Looks like we may need a git rebase -- this change was done a few weeks ago, out of scope of this PR

File: bigbluebutton-html5/.meteor/versions [hunk 2]
```diff
@@ -71,7 +71,7 @@ standard-minifier-js@2.8.0
 static-html@1.3.2
 templating-tools@1.2.1
 tracker@1.2.0
-typescript@4.4.1
+typescript@4.5.4
```
Comment (antobinary) [inline bigbluebutton-html5/.meteor/versions:74]: Let's try to upgrade React before this PR, so these changes are minimized

File: bigbluebutton-html5/package.json [hunk 3]
```diff
@@ -51,22 +53,25 @@
...
     "meteor-node-stubs": "^1.2.1",
     "postcss-nested": "^5.0.6",
+    "mobx": "6.4.2",
     "probe-image-size": "^4.1.1",
     "prom-client": "^13.2.0",
     "prop-types": "^15.8.1",
     "queue": "^6.0.2",
     "re-resizable": "^4.11.0",
-    "react": "^16.14.0",
+    "react": "^17.0.2",
```
Comment (antobinary) [inline bigbluebutton-html5/package.json:62]: Let's try to upgrade React before this PR, so these changes are minimized

File: package-lock.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "lockfileVersion": 1
```
Comment (antobinary) [inline package-lock.json:2]: This file can be removed (I believe)

File: bigbluebutton-html5/imports/api/annotations/server/methods.js [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
+import sendAnnotations from './methods/sendAnnotations';
 import sendBulkAnnotations from './methods/sendBulkAnnotations';
+import deleteAnnotations from './methods/deleteAnnotations';
 
 Meteor.methods({
-  undoAnnotation,
   clearWhiteboard,
-  sendAnnotation,
+  sendAnnotations,
+  sendAnnotations,
```
Comment (antobinary) [inline bigbluebutton-html5/imports/api/annotations/server/methods.js:10]:
```suggestion
```

duplicated line

File: bigbluebutton-html5/package.json [hunk 6]
```diff
@@ -51,22 +53,25 @@
     "makeup-screenreader-trap": "0.0.5",
     "meteor-node-stubs": "^1.2.1",
     "postcss-nested": "^5.0.6",
+    "mobx": "6.4.2",
```
[the remark this answers is not in the document]
Comment (KDSBrowne) [reply bigbluebutton-html5/package.json:56]: [mobx](https://github.com/mobxjs/mobx) is the state manager used in `TLDraw`.

File: bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx [hunk 7]
```diff
@@ -113,27 +113,27 @@ export default function Whiteboard(props) {
...
+            point: [50, 60],
+            isLocked: true,
+            size: [2560 / 3.5, 1440 / 3.5],
+            style: {
+              dash: DashStyle.Draw,
+              size: SizeStyle.Medium,
+              color: ColorStyle.Blue,
+            },
+          };
+        } catch (err) {
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx:133]: Should we log here or... ?

**Review by ramonlsouza — CHANGES_REQUESTED** [review]
- if a user leaves the whiteboard area, the cursor is not removed

https://user-images.githubusercontent.com/3728706/168598623-d701281b-724e-42d0-9c79-4af29e1534ec.mp4

- whiteboard size is different than slide size (slide does not use all available space)
slide-size

- can not move after zooming (pan tool)

https://user-images.githubusercontent.com/3728706/168599375-82416674-0947-454a-a487-80e04c77bfa4.mp4

- zoom % is not kept if user refreshes page

https://user-images.githubusercontent.com/3728706/168599524-b3a0ad69-8eba-4892-b84c-7d14df8bfa8b.mp4

- bbb crashes (for other users) if presenter refreshes

https://user-images.githubusercontent.com/3728706/168599621-106e929f-58ce-4bf8-851d-4dd2781d9c40.mp4

- “move to page” (appears when an annotation is selected) does not work as expected

https://user-images.githubusercontent.com/3728706/168599711-3e1f98cd-8a19-40a2-845b-2a7b076c37c5.mp4

**Comment by KDSBrowne** [comment]
> can not move after zooming (pan tool)

I think this is a usability issue. The functionality is a little different in tldraw because there is no tool to indicate being in a pan mode. The functionality does appear to work 
TLDRAW-PAN_ZOOM

**Review by ramonlsouza — APPROVED** [review]

**Pull request merged**